File: Codes/2274-maximum-consecutive-floors-without-special-floors.cpp
/*
 * @Author: LetMeFly
 * @Date: 2025-01-06 20:32:56
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-01-06 20:35:34
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

class Solution {
public:
    int maxConsecutive(int bottom, int top, vector<int>& special) {
        sort(special.begin(), special.end());
        int ans = top - special.back();
        bottom--;
        for (int t : special) {
            ans = max(ans, t - bottom - 1);
            bottom = t;
        }
        return ans;
    }
};

File: Codes/2274-maximum-consecutive-floors-without-special-floors.py
'''
Author: LetMeFly
Date: 2025-01-06 20:36:27
LastEditors: LetMeFly.xyz
LastEditTime: 2025-01-06 20:38:18
'''
from typing import List

class Solution:
    def maxConsecutive(self, bottom: int, top: int, special: List[int]) -> int:
        special.sort()
        ans = max(special[0] - bottom, top - special[-1])
        for i in range(1, len(special)):
            ans = max(ans, special[i] - special[i - 1] - 1)
        return ans